core-plugins/src/main/java/io/cdap/plugin/batch/joiner/Joiner.java

```diff
@@ -263,4 +281,24 @@ public void prepareRun(BatchJoinerContext context) {
 
     return operations;
   }
+
+  /**
+   * Reorders join stages so the supplied stage name is always first.
+   * @param stages list of input stages
+   * @param leftStage stage to move to the first position in the input list.
+   */
+  protected void reorderJoinStages(List<JoinStage> stages, String leftStage) {
+    stages.sort((js1, js2) -> {
```

Comment:
Sorting the list is an odd way to do this. Is there a reason we aren't just swapping the first stage with `leftStage` using something like `lastIndexOf(leftStage)` to find the index where the swap needs to happen?